internal/indexer/indexer.go

```diff
@@ -337,30 +372,9 @@ func (i *Indexer) positionAndDocument(p *packages.Package, pos token.Pos) (token
 	return position, d, true
 }
 
-// markRange sets an empty range identifier in the ranges map for the given position.
-// If a  range for this identifier has already been marked, this method returns false.
-func (i *Indexer) markRange(pos token.Position) bool {
-	i.stripedMutex.RLockKey(pos.Filename)
-	_, ok := i.ranges[pos.Filename][pos.Offset]
-	i.stripedMutex.RUnlockKey(pos.Filename)
-	if ok {
-		return false
-	}
-
-	i.stripedMutex.LockKey(pos.Filename)
-	defer i.stripedMutex.UnlockKey(pos.Filename)
-
-	if _, ok := i.ranges[pos.Filename][pos.Offset]; ok {
-		return false
-	}
-
-	i.ranges[pos.Filename][pos.Offset] = 0 // placeholder
-	return true
-}
-
 // indexDefinition emits data for the given definition object.
 func (i *Indexer) indexDefinition(p *packages.Package, filename string, document *DocumentInfo, pos token.Position, obj types.Object, typeSwitchHeader bool, ident *ast.Ident) uint64 {
-	rangeID := i.emitter.EmitRange(rangeForObject(obj, pos))
+	rangeID := i.ensureRangeFor(pos, obj)
```

Comment:
Because the range may have already been emitted in the `indexSymbols` stage.

Reply:
answers: https://github.com/sourcegraph/lsif-go/pull/142